File: Python/time_comparison.py
import time
from concurrent_multiply import concurrent_multiplication1, concurrent_multiplication2
from sequential_multiply import sequential_multiplication


def time_comparison(matrix1, matrix2):
    start = time.time()
    sequential_multiplication(matrix1, matrix2)
    end = time.time()
    print("Sequential:", end - start)

    start = time.time()
    concurrent_multiplication1(matrix1, matrix2)
    end = time.time()
    print("Using concurrent.futures:", end - start)

    start = time.time()
    concurrent_multiplication2(matrix1, matrix2)
    end = time.time()
    print("Using multiprocessing:", end - start)

File: C++/MatrixMultiplication/src/time_comparison.cpp
#include "time_comparison.hpp"
#include "sequential_multiply.hpp"
#include "concurrent_multiply.hpp"
#include <iostream>
#include <chrono>

void TimeComparison::compare(const std::vector<std::vector<int>> &matrix1, const std::vector<std::vector<int>> &matrix2)
{
  auto start = std::chrono::high_resolution_clock::now();
  SequentialMultiply::multiply(matrix1, matrix2);
  auto end = std::chrono::high_resolution_clock::now();
  std::chrono::duration<double, std::milli> sequentialTime = end - start;
  std::cout << "Sequential time: " << sequentialTime.count() << "ms\n";

  start = std::chrono::high_resolution_clock::now();
  ConcurrentMultiply::multiply(matrix1, matrix2);
  end = std::chrono::high_resolution_clock::now();
  std::chrono::duration<double, std::milli> concurrentTime = end - start;
  std::cout << "Concurrent time: " << concurrentTime.count() << "ms\n";
}
